src/qt/guiutil_font.cpp

```diff
@@ -0,0 +1,596 @@
+
+    // QPointer becomes nullptr for objects that were deleted.
+    // Remove them from mapDefaultFontSize and mapFontUpdates
+    // before proceeding any further.
+    size_t nRemovedDefaultFonts{0};
+    auto itd = mapWidgetDefaultFontSizes.begin();
+    while (itd != mapWidgetDefaultFontSizes.end()) {
+        if (itd->first.isNull()) {
+            itd = mapWidgetDefaultFontSizes.erase(itd);
+            ++nRemovedDefaultFonts;
+        } else {
+            ++itd;
+        }
+    }
+
+    size_t nRemovedFontUpdates{0};
+    auto itn = mapFontUpdates.begin();
+    while (itn != mapFontUpdates.end()) {
+        if (itn->first.isNull()) {
+            itn = mapFontUpdates.erase(itn);
+            ++nRemovedFontUpdates;
+        } else {
+            ++itn;
+        }
+    }
+
+    size_t nUpdatable{0}, nUpdated{0};
+    std::map<QWidget*, QFont> mapWidgetFonts;
+    // Loop through all widgets
+    for (QWidget* w : qApp->allWidgets()) {
+        std::vector<QString> vecIgnoreClasses{
+            "QWidget", "QDialog", "QFrame", "QStackedWidget", "QDesktopWidget", "QDesktopScreenWidget",
+            "QTipLabel", "QMessageBox", "QMenu", "QComboBoxPrivateScroller", "QComboBoxPrivateContainer",
+            "QScrollBar", "QListView", "BitcoinGUI", "WalletView", "WalletFrame", "QVBoxLayout", "QGroupBox"
+        };
+        std::vector<QString> vecIgnoreObjects{
+            "embeddedFont_label_1", "embeddedFont_label_9", "messagesWidget", "systemFont_label_1", "systemFont_label_9"
+        };
+        if (std::find(vecIgnoreClasses.begin(), vecIgnoreClasses.end(), w->metaObject()->className()) != vecIgnoreClasses.end() ||
+            std::find(vecIgnoreObjects.begin(), vecIgnoreObjects.end(), w->objectName()) != vecIgnoreObjects.end()) {
+            continue;
+        }
+        ++nUpdatable;
+
+        QFont font = w->font();
+        assert(font.pointSize() > 0);
+        font.setFamily(qApp->font().family());
+        font.setWeight(getFontWeightNormal());
+        font.setStyleName(qApp->font().styleName());
+        font.setStyle(qApp->font().style());
+
+        // Insert/Get the default font size of the widget
+        auto itDefault = mapWidgetDefaultFontSizes.emplace(w, font.pointSize());
+
+        auto it = mapFontUpdates.find(w);
+        if (it != mapFontUpdates.end()) {
+            int nSize = it->second.size;
+            if (nSize == -1) {
+                nSize = itDefault.first->second;
+            }
+            font = getFont(it->second.family, toQFontWeight(it->second.weight), it->second.is_italic, nSize);
+        } else {
+            font.setPointSizeF(getScaledFontSize(itDefault.first->second));
+        }
+
+        if (w->font() != font) {
+            auto itWidgetFont = mapWidgetFonts.emplace(w, font);
+            assert(itWidgetFont.second);
+            ++nUpdated;
+        }
+    }
+    qDebug().nospace() << __func__ << " - widget counts: updated/updatable/total(" << nUpdated << "/" << nUpdatable
```

Comment:
nit: consider using here strprintf() to increase readability

Reply:
See [comment](https://github.com/dashpay/dash/pull/6831#discussion_r2651760476)